idseq_pipeline/commands/host_filtering_functions.py

```diff
@@ -236,39 +236,37 @@ def run_star_part(output_dir, genome_dir, fastq_files):
     execute_command_realtime_stdout(" ".join(star_command_params), os.path.join(output_dir, "Log.progress.out"))
 
 def run_star(fastq_files):
-    # check if genome downloaded already
-    genome_file = os.path.basename(STAR_GENOME)
-    if not os.path.isfile("%s/%s" % (REF_DIR, genome_file)):
-        execute_command("aws s3 cp %s %s/" % (STAR_GENOME, REF_DIR))
-        execute_command("cd %s; tar xvfz %s" % (REF_DIR, genome_file))
+    star_outputs = [STAR_OUT1, STAR_OUT2]
+    num_fastqs = len(fastq_files)
+    def unmapped_files_in(some_dir):
+        return ["%s/Unmapped.out.mate%d" % (some_dir, i+1) for i in range(num_fastqs)]
+    genome_name = os.path.basename(STAR_GENOME).rstrip(".gz").rstrip(".tar")
+    if genome_name != "STAR_genome":
+        with print_lock:
+            print "Oh hello interesting new genome {}".format(genome_name)
+    genome_dir = os.path.join(REF_DIR, genome_name)
+    if not os.path.exists(genome_dir):
```

Comment:
this could potentially be an issue if different hosts are sharing the same genome dir

Reply:
yes, it could!   i am happy to address that issue and also switch us into an environment where that sharing can happen;   and by different hosts i think realistically what we mean here is different jobs on the same instance --- because if you mean using a shared filesystem on amazon, I benchmarked EFS and it's worthless

Reply:
this said, it should be handle separately throughout the pipeline

i am going to refactor quite a bit to make it possible to address those issues

i only looked into this first because of the terrible performance of downloading 25GB index;   then found the correctness bugs that had to be fixed